### DISCUSSION TITLE
Merge snapshot related  CLI options/flags

### DISCUSSION THREAD

**Discussion by CaroFG** [meilisearch/product#566] [Feedback & Feature Proposal]
Related to #536 

The are 2 CLI option / flags to schedule snapshots:

- `--schedule-snapshot`  doesn't take any value and activates snapshot creation 
- `--snapshot-interval-sec` takes an argument and defines the interval between each snapshot

They could be merged into one CLI option: '--schedule-snapshots' which would activate snapshots and set the interval. It would have a default value that could be changed by giving an argument.

**Comment by gmourier** [comment]
Thanks, @CaroFG.

To recap, your proposal would be to:

1. Remove `--snapshot-interval-sec`
2. `--schedule-snapshot` now accepts an interval in seconds, if `--schedule-snapshot` is given, snapshot are activated

I'm not sure if it's possible to have a default value in that case. @Kerollmops WDYT?

**Reply by curquiza** [reply]
My two cents: docs team should still provide a recommended value of interval of the users (the current default one?)

**Reply by Kerollmops** [reply]
It looks like it is possible to do that with clap. However, I don't know if you want to prioritize that work.

```rust
use clap::Parser;

#[derive(Debug, Parser)]
struct Opt {
    #[arg(long, default_missing_value = "42")]
    schedule_snapshot: Option<Option<u32>>,
}

fn main() {
    dbg!(Opt::parse());
}
```

```bash
cargo run -- # gives None
cargo run -- --schedule-snapshot # gives Some(Some(42))
cargo run -- --schedule-snapshot 32 # gives Some(Some(32))
```

**Reply by gmourier** [reply]
Thanks, @Kerollmops. It's part of the CLI stabilization work for v1.0; if it's quick to do it, let's go. There is no need to change the current default value `86400` (24h).

**Reply by Kerollmops** [reply]
Will you update the spec for v1.0? It looks to me that it is simple to do.

**Reply by gmourier** [reply]
Could you do the spec at the same time you do the change for v1.0? 🙇‍♂️

**Reply by curquiza** [reply]
Issue created: https://github.com/meilisearch/meilisearch/issues/3131

**Comment by gmourier** [comment]
Discussed at the weekly meeting and with @dureuill

If it's too long or complicated to implement, we stop here and keep the two current options because the expected impact is not significant enough to invest too much in it.

**Comment by gmourier** [comment]
Hello 👋

Locking that discussion since it will be released with `v1` next week.

Please open a new discussion if you have some enhancements to propose! 

Thank you @CaroFG 😇
